### PR TITLE
PIM-7785-7786: upload/download a record media file

### PR DESCRIPTION
**Pull request by NolwennP**

### CODE REVIEW

File: content/swagger/resources/reference_entity_media_files/routes/reference_entity_media_files_code.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+get:
```
Comment (ahocquard) [inline content/swagger/resources/reference_entity_media_files/routes/reference_entity_media_files_code.yaml]: we should avoid to expose this endpoint as it's useless.

I think it's better to return the location in the header:
Location: http:/...../codeofthefile
Reference-entities-media-file-code: ....

Comment (NolwennP) [reply content/swagger/resources/reference_entity_media_files/routes/reference_entity_media_files_code.yaml]: Done!
Now the get endpoint download directly the file

File: content/rest-api/resources.md [hunk 2]
```diff
@@ -1079,11 +1079,54 @@ Below is the JSON standard format representing a published product. Notice how t
...
 
+### Reference entity (3.x and EE only)
+
+Reference entities have been introduced in the PIM in the 3.0 version.
+
+In the Akeneo UI, you can find the reference entities in the `Entities` menu. Below is an example of a reference entity in the UI.
+
+![Reference entity](/img/screenshots/...png)
+
+Below is the JSON standard format representing a reference entity.
```
Comment (ahocquard) [inline content/rest-api/resources.md]:
```suggestion
Below is the JSON format representing a reference entity.
```

File: content/rest-api/resources.md [hunk 3]
```diff
@@ -1279,3 +1322,15 @@ Note that the `locale` and `channel` properties are all set to `null` in this ca
 ::: panel-link Want to update reference entity record values? [Here you go!](/documentation/update.html#patch-reference-entity-record-values)
 :::
 
+### Record media file (3.x and EE only)
+
+Record media files corresponds to the images that you can link to the records of your reference entities.
+
+In the Akeneo UI, you can find these media files in the detail of a record. In the screenshot below, there are two record media files:
```
Comment (ahocquard) [inline content/rest-api/resources.md]:
```suggestion
In the Akeneo UI, you can find these media files in the details of a record. In the screenshot below, there are two record media files: 
```

File: content/swagger/resources/reference_entities/routes/reference_entities.yaml [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+      examples: {
+        _links: {
+          self: {
+            href: "https://demo.akeneo.com/api/rest/v1/reference-entities?limit=3"
+          },
+          first: {
+            href: "https://demo.akeneo.com/api/rest/v1/reference-entities?limit=3"
+          },
+          next: {
+            href: "https://demo.akeneo.com/api/rest/v1/reference-entities?page=3&search_after=2x055w%3D%3D"
```
Comment (ahocquard) [inline content/swagger/resources/reference_entities/routes/reference_entities.yaml]:
```suggestion
            href: "https://demo.akeneo.com/api/rest/v1/reference-entities?limit=3&search_after=2x055w%3D%3D"
```

File: content/rest-api/resources.md (hunk 3 above)
Comment (NolwennP) [reply content/rest-api/resources.md]: Nope!

File: content/swagger/resources/reference_entities/routes/reference_entities_code.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  x-versions:
+    - "3.0"
+  x-ee: true
+  description: This endpoint allows you to get the information about a given reference entity.
+  parameters:
+    - $ref: '#/parameters/code'
+  responses:
+    200:
+      description: OK
+      x-details: Returns the content of the reference entity in JSON standard format.
```
Comment (ahocquard) [inline content/swagger/resources/reference_entities/routes/reference_entities_code.yaml]:
```suggestion
      x-details: Returns the content of the reference entity in JSON format.
```

File: content/swagger/resources/reference_entity_media_files/routes/reference_entity_media_files.yaml [hunk 6]
```diff
@@ -0,0 +1,34 @@
+post:
+  summary: Create a new record media file
+  operationId: "post_reference_entity_media_files"
+  description: This endpoint allows you to create a new media file and associate it to the main image or to an attribute value of a record of reference entity.
```
Comment (ahocquard) [inline content/swagger/resources/reference_entity_media_files/routes/reference_entity_media_files.yaml]:
```suggestion
  description: This endpoint allows you to create a new media file and associate it to the main image or to an attribute value of a record of a reference entity.
```

File: content/swagger/resources/reference_entity_media_files/routes/reference_entity_media_files.yaml [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+      required: true
+    - name: body
+      in: body
+      x-form-data: true
+      schema:
+        required: ["file"]
+        properties:
+          file:
+            type: string
+            description: The binaries of the record media file
```
Comment (ahocquard) [inline content/swagger/resources/reference_entity_media_files/routes/reference_entity_media_files.yaml]:
```suggestion
            description: The binary of the record media file
```

**Pull request merged**